crates/soft-confirmation-rule-enforcer/src/hooks.rs

```diff
@@ -34,13 +34,9 @@ where
         // Adding one more l2 block will exceed the max L2 blocks per L1
         if l2_block_count + 1 > max_l2_blocks_per_l1 {
             // block count per l1 block should not be more than max L2 blocks per L1
-            return Err(
-                ApplySoftConfirmationError::TooManySoftConfirmationsOnDaSlot {
-                    hash: da_root_hash,
-                    sequencer_pub_key: soft_confirmation_info.sequencer_pub_key().to_vec(),
-                    max_l2_blocks_per_l1,
-                },
-            );
+            return Err(SoftConfirmationError::Other(
```

Comment:
Why string is used instead of ApplySoftConfirmationError::TooManySoftConfirmationsOnDaSlot?

Reply:
I thought normal fields of the Error enum would be used by the stf itself and the Other(String) error would be used by modules.

Reply:
So that if in the future we add any module functionality, or we add any new modules, we don't add errors to the enum but use the Other field.

I can change it tho.

Reply:
We may make it a separate issue

Reply:
https://github.com/chainwayxyz/citrea/issues/1048